jupyter_notebook/auth/login.py

```diff
@@ -37,7 +37,8 @@ def post(self):
         typed_password = self.get_argument('password', default=u'')
         if self.login_available(self.settings):
             if passwd_check(self.hashed_password, typed_password):
-                self.set_secure_cookie(self.cookie_name, str(uuid.uuid4()))
+                self.set_secure_cookie(self.cookie_name, str(uuid.uuid4()), 
+                        secure=self.settings.get('secure_cookie', (self.request.protocol == 'https'))
```

Comment:
as [discovered on JupyterHub](https://github.com/jupyter/jupyterhub/pull/231#issuecomment-93862873), `secure=False` doesn't actually work. The `secure` kwarg can only be specified when it should be True (a bug in Tornado, fixed in 4.2)